etna/core/fields.py

```diff
@@ -153,14 +187,24 @@ def clean(self, value):
 
         if errors:
             # Raise collected errors all at once
-            raise ValidationError(errors)
+            # Make unique message when using same error message in validators
+            m = []
+            e = []
+            for error in errors:
+                if isinstance(error, str):
+                    msg = error
+                elif isinstance(error, ValidationError):
+                    msg = error.message
+                if msg not in m:
+                    m.append(msg)
+                    e.append(error)
+            raise ValidationError(e)
 
         try:
             out = self.compress(clean_data)
         except ValueError:
-            # The specified day must be invalid for the month, so add an error message
-            # to the 'day' sub-field widget before raising
-            msg = "Enter a valid day for the month you entered"
+            # When Day, Month belong to correct range and Year is 0, then error-value "year 0 is out of range"
+            msg = "Entered date must be a real date, for example 23 9 2017."
```

Comment:
What's the benefit of making this generic? Surely that just makes it difficult to see what is wrong?

Reply:
This is the requirement for those error handling.